packages/tauri/generate-tauri-update-file.js

```diff
@@ -0,0 +1,41 @@
+const { resolve } = require("path");
+const { readFileSync, writeFileSync } = require("fs");
+
+const tauriUpdateFilePath = resolve(__dirname, "tauri-update.json");
+const packageFilePath = resolve(__dirname, "package.json");
+
+const tauriUpdate = {
+    name: "[VERSION]",
+    pub_date: "[DATE]",
+    platforms: {
+        darwin: {
+            signature: "[URL]",
+            url: "[URL]",
+        },
+        linux: {
+            signature: "[URL]",
+            url: "[URL]",
+        },
+        win64: {
+            signature: "[URL]",
+            url: "[URL]",
+        },
+    },
+};
+
+const packageFileContents = readFileSync(packageFilePath, "utf-8");
+const package = JSON.parse(packageFileContents);
+
+const { version } = package;
+const now = new Date().toISOString();
+
+tauriUpdate.name = `v${version}`;
+tauriUpdate.pub_date = now;
+tauriUpdate.platforms.darwin.url = `https://github.com/padloc/padloc/releases/download/v${version}/Padloc.app.tar.gz`;
+tauriUpdate.platforms.darwin.signature = `${tauriUpdate.platforms.darwin.url}.sig`;
+tauriUpdate.platforms.linux.url = `https://github.com/padloc/padloc/releases/download/v${version}/padloc_${version}_amd64.AppImage.tar.gz`;
+tauriUpdate.platforms.linux.signature = `${tauriUpdate.platforms.linux.url}.sig`;
+tauriUpdate.platforms.win64.url = `https://github.com/padloc/padloc/releases/download/v${version}/Padloc_${version}_x64.msi.zip`;
+tauriUpdate.platforms.win64.signature = `${tauriUpdate.platforms.win64.url}.sig`;
```

Comment:
The main advantage here is that I can reuse the objects (like `${tauriUpdate.platforms.linux.url}.sig`), otherwise we'll have to duplicate the repo URL. Both are fine, it's mostly so we have a "template" to start with that we populate, which I think makes things clearer. I'm fine to do it either way, here. So, your call.

Reply:
If you're worried about repetition, you could simply store the "base url" in a variable. For me, the "canonical form" of the whole thing would look something like this:

```js
const baseUrl = `https://github.com/padloc/padloc/download/v${version}/`
const tauriUpdate = {
    name: version,
    pub_date: new Date(),
    platforms: {
        darwin: {
            url: `${baseUrl}/Padloc.app.tar.gz`,
            signature: `${baseUrl}/Padloc.app.tar.gz.sig`,
        },
        linux: {
            url: `${baseUrl}/padloc_${version}_amd64.AppImage.tar.gz`,
            signature: `${baseUrl}/padloc_${version}_amd64.AppImage.tar.gz.sig`,
        },
        win64: {
            url: `${baseUrl}/Padloc_${version}_x64.msi.zip`,
            signature: `${baseUrl}/Padloc_${version}_x64.msi.zip.sig`,
        },
    },
};
```

But that's just my preference and I feel like we're getting dangerously close to shedding bikes again (especially since this is basically just a "support script"), so I'm totally fine with keeping it as-is 😊

Reply:
I'll do it your way (except for `new Date()`, because I want to make sure it's clear what the expectation is, and if the object would be passed around before "stringified", it could be an `int` instead of `string`, and I don't like to promote that JS inconsistency).